airflow/models/taskinstance.py

```diff
@@ -2558,12 +2508,6 @@ def _check_and_change_state_before_execution(
 
         if isinstance(task_instance, TaskInstance):
```

Comment:
Is that `if` still needed if there is no else?

Reply:
Good point . No.